assets/global.js

```diff
@@ -443,6 +443,7 @@ class ModalDialog extends HTMLElement {
       'click',
       this.hide.bind(this, false)
     );
+    ['shopify:block:select', 'shopify:section:select'].forEach(event => document.addEventListener(event, () => this.hide()));
```

Comment:
We could try and target a few more modals: 
- details-modal (comes up when you click the search button in the header)
- password-modal which is based off of `details-modal` (which shows the password prompt on the password page)

[Here is for the search in the header](https://screenshot.click/08-10-bbgey-lgs1v.mp4) which feels maybe like a niche scenario. 

And [here is the password page](https://screenshot.click/08-20-djzur-2ae4q.mp4) which is a bit more annoying.

Reply:
Since the `PasswordModal` inherits from `DetailsModal`, adding the editor events in the `DetailsModal` should fix both of those issues. It's worth finding out why the password modal won't open in the first place and whether that's a bug, so I think it would be better to look at all of that as part of another issue.